### PR TITLE
Migrate party details screen to voyager

### PR DESCRIPTION
**Pull request by gmatyszczak**

### CODE REVIEW

File: app/src/main/kotlin/com/intive/picover/main/navigation/view/PicoverNavHost.kt [hunk 1]
```diff
@@ -60,7 +61,8 @@ fun PicoverNavHost(
 				route = "partyDetails/{partyId}",
 				arguments = listOf(navArgument("partyId") { type = NavType.StringType }),
 			) {
-				PartyDetailsScreen(hiltViewModel())
+				val partyId = it.arguments!!.getString("partyId")!!
+				Navigator(PartyDetailsScreen(partyId))
```
Comment (gmatyszczak) [inline app/src/main/kotlin/com/intive/picover/main/navigation/view/PicoverNavHost.kt:65]: Plan is to migrate each child screen to voyager first and after that migrate whole main screen to TabNavigator from voyager, we'll see how it goes 😄 So temporarily each child is wrapped in it's own Navigator to keep all state restoration and view model disposing working correctly

File: app/src/main/kotlin/com/intive/picover/parties/di/PartiesModule.kt [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+import dagger.multibindings.IntoMap
+
+@Module
+@InstallIn(ActivityComponent::class)
+abstract class PartiesModule {
+
+	@Binds
+	@IntoMap
+	@ScreenModelFactoryKey(PartyDetailsViewModel.Factory::class)
+	abstract fun bindPartyDetailsViewModelFactory(factory: PartyDetailsViewModel.Factory): ScreenModelFactory
```
Comment (gmatyszczak) [inline app/src/main/kotlin/com/intive/picover/parties/di/PartiesModule.kt:19]: Internally there's a map of ScreenModelFactory in voyager, if we have a view model with arguments we need to create an AssistedFactory for it, and bind this factory into that internal map. Perhaps in the future we could write some custom KSP processor to generate that module for us

https://github.com/adrielcafe/voyager/blob/main/samples/android/src/main/java/cafe/adriel/voyager/sample/hiltIntegration/HiltModule.kt

File: app/src/main/kotlin/com/intive/picover/parties/viewmodel/PartyDetailsViewModel.kt [hunk 3]
```diff
@@ -1,40 +1,47 @@
...
+import dagger.assisted.AssistedFactory
+import dagger.assisted.AssistedInject
 import kotlinx.coroutines.flow.catch
+import kotlinx.coroutines.flow.update
 import kotlinx.coroutines.launch
 
-@HiltViewModel
-class PartyDetailsViewModel @Inject constructor(
-	savedStateHandle: SavedStateHandle,
+class PartyDetailsViewModel @AssistedInject constructor(
```
Comment (gmatyszczak) [inline app/src/main/kotlin/com/intive/picover/parties/viewmodel/PartyDetailsViewModel.kt:17]: I kept it as "ViewModel" for now to not change too much, but we can decide on naming later

File: app/src/main/kotlin/com/intive/picover/parties/di/PartiesModule.kt (hunk 2 above)
Comment (MarekMacko) [reply app/src/main/kotlin/com/intive/picover/parties/di/PartiesModule.kt:19]: Is it possibile to have default binding for `SavedStateHandle` in view model?

Comment (gmatyszczak) [reply app/src/main/kotlin/com/intive/picover/parties/di/PartiesModule.kt:19]: There will be no `SavedStateHandle` 😄

Comment (MarekMacko) [reply app/src/main/kotlin/com/intive/picover/parties/di/PartiesModule.kt:19]: Oh... right 😄  I'm still tied to android world 😄

File: app/src/main/kotlin/com/intive/picover/parties/model/PartyDetailsState.kt [hunk 4]
```diff
@@ -0,0 +1,8 @@
+package com.intive.picover.parties.model
+
+import com.intive.picover.common.viewmodel.state.MVIStateType
+
+data class PartyDetailsState(
+	val type: MVIStateType = MVIStateType.LOADING,
+	val party: Party? = null,
+)
```
Comment (MarekMacko) [inline app/src/main/kotlin/com/intive/picover/parties/model/PartyDetailsState.kt:8]: We could look for other convenient way to provide initial state, just to not use default arguments for all properties what leads to exposing empty constructor only for this purpose. 
I didn't find it yet but I wanted just to mention that 😄

**Comment by MarekMacko** [comment]
Nice! 🍻

**Review by MarekMacko — APPROVED** [review]

**Pull request merged**